From cb45afd637934918c91d498c84e771e0b8eb794f Mon Sep 17 00:00:00 2001
From: Alicia Codex <alicia@schlaile.de>
Date: Sun, 29 Mar 2026 16:31:55 +0200
Subject: [PATCH] PHP 8.5: Avoid null viewer PHID in Calendar event search

---
 .../query/PhabricatorCalendarEventSearchEngine.php       | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/applications/calendar/query/PhabricatorCalendarEventSearchEngine.php b/src/applications/calendar/query/PhabricatorCalendarEventSearchEngine.php
index 1631a888ed..65f57bd47c 100644
--- a/src/applications/calendar/query/PhabricatorCalendarEventSearchEngine.php
+++ b/src/applications/calendar/query/PhabricatorCalendarEventSearchEngine.php
@@ -20,9 +20,14 @@ final class PhabricatorCalendarEventSearchEngine
    */
   public function newQuery() {
     $viewer = $this->requireViewer();
+    $viewer_phid = $viewer->getPHID();
+
+    $query = new PhabricatorCalendarEventQuery();
+    if ($viewer_phid) {
+      $query->needRSVPs(array($viewer_phid));
+    }
 
-    return id(new PhabricatorCalendarEventQuery())
-      ->needRSVPs(array($viewer->getPHID()));
+    return $query;
   }
 
   protected function shouldShowOrderField() {
-- 
2.47.3

