diff --git a/resources/sql/autopatches/20170528.maniphestdupes.php b/resources/sql/autopatches/20170528.maniphestdupes.php
index aea8ae93d2..abbf2857fc 100644
--- a/resources/sql/autopatches/20170528.maniphestdupes.php
+++ b/resources/sql/autopatches/20170528.maniphestdupes.php
@@ -1,5 +1,30 @@
 <?php
 
+$tables = array(
+  new PhabricatorWorkerActiveTask(),
+  new PhabricatorWorkerArchiveTask(),
+);
+
+foreach ($tables as $table) {
+  $conn = $table->establishConnection('w');
+  $table_name = $table->getTableName();
+
+  $has_column = queryfx_one(
+    $conn,
+    'SHOW COLUMNS FROM %T LIKE %>',
+    $table_name,
+    'containerPHID');
+
+  if ($has_column) {
+    continue;
+  }
+
+  queryfx(
+    $conn,
+    'ALTER TABLE %T ADD containerPHID VARBINARY(64)',
+    $table_name);
+}
+
 $table = new ManiphestTransaction();
 $add_edges = array();
 
diff --git a/src/infrastructure/storage/patch/PhabricatorBuiltinPatchList.php b/src/infrastructure/storage/patch/PhabricatorBuiltinPatchList.php
index 5cc7da76c3..04e08db9ca 100644
--- a/src/infrastructure/storage/patch/PhabricatorBuiltinPatchList.php
+++ b/src/infrastructure/storage/patch/PhabricatorBuiltinPatchList.php
@@ -41,9 +41,9 @@ final class PhabricatorBuiltinPatchList extends PhabricatorSQLPatchList {
     $auto_root = $root.'/resources/sql/autopatches/';
     $autopatches = $this->buildPatchesFromDirectory($auto_root);
 
-    $autopatches['20170528.maniphestdupes.php']['after'] = array(
-      '20210122.queuecontainer.01.sql',
-    );
+    $autopatches['20210122.queuecontainer.01.sql']['type'] = 'php';
+    $autopatches['20210122.queuecontainer.01.sql']['name'] =
+      $this->getPatchPath('20210122.queuecontainer.php');
 
     $patches += $autopatches;
 
diff --git a/resources/sql/patches/20210122.queuecontainer.php b/resources/sql/patches/20210122.queuecontainer.php
new file mode 100644
index 0000000..89e1e5b
--- /dev/null
+++ b/resources/sql/patches/20210122.queuecontainer.php
@@ -0,0 +1,26 @@
+<?php
+
+$tables = array(
+  new PhabricatorWorkerActiveTask(),
+  new PhabricatorWorkerArchiveTask(),
+);
+
+foreach ($tables as $table) {
+  $conn = $table->establishConnection('w');
+  $table_name = $table->getTableName();
+
+  $has_column = queryfx_one(
+    $conn,
+    'SHOW COLUMNS FROM %T LIKE %>',
+    $table_name,
+    'containerPHID');
+
+  if ($has_column) {
+    continue;
+  }
+
+  queryfx(
+    $conn,
+    'ALTER TABLE %T ADD containerPHID VARBINARY(64)',
+    $table_name);
+}
